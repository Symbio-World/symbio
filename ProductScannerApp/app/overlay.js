import React from 'react';
import {
  Text,
  View,
} from 'react-native';

export const Overlay = (barcode) => (
  <View
    key={barcode.data}
    style={{
      position: 'absolute',
      zIndex: 1,
      top: barcode.top,
      left: barcode.left,
      borderRadius: 5,
      backgroundColor: '#00ff00'
    }}
  >
    <Text>
      {barcode.data}
    </Text>
  </View>
)