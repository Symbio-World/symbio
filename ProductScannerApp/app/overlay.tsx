import React from 'react'
import { Modal, TouchableWithoutFeedback, View } from 'react-native'
import { t } from 'react-native-tailwindcss'

type Props = {
  children?: React.ReactNode
  onDismiss?: () => void
}

export const Overlay: React.FC<Props> = ({ children, onDismiss }) => (
  <Modal
    transparent={true}
    visible={true}
    animationType="none"
    onRequestClose={onDismiss}
  >
    <TouchableWithoutFeedback onPress={onDismiss}>
      <View style={[t.absolute, t.inset0, t.bgTransparent]} />
    </TouchableWithoutFeedback>

    <View style={[t.flex1, t.m12]}>{children}</View>
  </Modal>
)
