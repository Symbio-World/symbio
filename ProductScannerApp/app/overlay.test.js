/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Overlay } from './overlay';

it('renders correctly', () => {
  renderer.create(<Overlay />);
});
