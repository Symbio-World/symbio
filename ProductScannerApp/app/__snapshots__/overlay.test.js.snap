// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "margin": 48,
        },
      ]
    }
  >
    <Text>
      test
    </Text>
  </View>
</Modal>
`;
